pkg/extensions/tbtc/recovery/storage.go

```diff
@@ -147,11 +149,28 @@ func (dis *DerivationIndexStorage) GetNextIndex(extendedPublicKey string) (uint3
 		return 0, err
 	}
 
-	index, err := dis.read(extendedPublicKey)
+	lastIndex, err := dis.read(extendedPublicKey)
 	if err != nil {
 		return 0, err
 	}
-	return uint32(index + 1), nil
+	nextIndex := uint32(lastIndex + 1)
```

Comment:
Maybe `startIndex`? `nextIndex` suggests to me that we will be increasing it iteratively in the loop.

```suggestion
	startIndex := uint32(lastIndex + 1)
```